Modules/Relationships/src/Relationships.ConsumerApi/Controllers/RelationshipTemplatesController.cs

```diff
@@ -31,19 +32,38 @@ public RelationshipTemplatesController(IMediator mediator, IOptions<ApplicationO
     [HttpGet("{id}")]
     [ProducesResponseType(typeof(HttpResponseEnvelopeResult<RelationshipTemplateDTO>), StatusCodes.Status200OK)]
     [ProducesError(StatusCodes.Status404NotFound)]
-    public async Task<IActionResult> GetById(string id, CancellationToken cancellationToken)
+    public async Task<IActionResult> GetById(string id, [FromQuery] byte[]? password, CancellationToken cancellationToken)
     {
-        var template = await _mediator.Send(new GetRelationshipTemplateQuery { Id = id }, cancellationToken);
+        var template = await _mediator.Send(new GetRelationshipTemplateQuery { Id = id, Password = password }, cancellationToken);
         return Ok(template);
     }
 
-    [HttpGet]
+    [HttpGet("Alt/Templates")] // todo: remove 'Alt/Templates' extension once the transport tests are updated
     [ProducesResponseType(typeof(PagedHttpResponseEnvelope<ListRelationshipTemplatesResponse>),
         StatusCodes.Status200OK)]
-    public async Task<IActionResult> GetAll([FromQuery] PaginationFilter paginationFilter,
-        [FromQuery] IEnumerable<string> ids, CancellationToken cancellationToken)
+    public async Task<IActionResult> GetAll([FromQuery] PaginationFilter paginationFilter, [FromQuery] string templates, CancellationToken cancellationToken)
     {
-        var request = new ListRelationshipTemplatesQuery(paginationFilter, ids);
+        var relationshipTemplateQueries = JsonSerializer.Deserialize<List<RelationshipTemplateQuery>>(templates);
+
+        var request = new ListRelationshipTemplatesQuery(paginationFilter, relationshipTemplateQueries);
+
+        request.PaginationFilter.PageSize ??= _options.Pagination.DefaultPageSize;
+
+        if (paginationFilter.PageSize > _options.Pagination.MaxPageSize)
+            throw new ApplicationException(GenericApplicationErrors.Validation.InvalidPageSize(_options.Pagination.MaxPageSize));
+
+        var template = await _mediator.Send(request, cancellationToken);
+        return Paged(template);
+    }
+
+    [HttpGet] // todo: remove this endpoint once the transport tests are updated
+    [ProducesResponseType(typeof(PagedHttpResponseEnvelope<ListRelationshipTemplatesResponse>),
+        StatusCodes.Status200OK)]
+    public async Task<IActionResult> GetAllWithIds([FromQuery] PaginationFilter paginationFilter, [FromQuery] string ids, CancellationToken cancellationToken)
```

Comment:
This solution is meant for backwards compatibility. It won't be removed for a few months. Until then, we need to support both ways under the same URL. 

So you can remove the todo comment. Instead, add the `DeprecatedAttribute` with a corresponding message.

Reply:
I am afraid I cannot have two endpoints with the same verb and same URL. I've added the `[Obsolete]` attribute.

Reply:
Then use one endpoint with two parameters.

Reply:
You mean pass both ids and templates to the same endpoint?